WordPress/Classes/ViewRelated/Jetpack/Branding/Banner/JPScrollViewDataDelegate.swift

```diff
@@ -0,0 +1,9 @@
+import Foundation
+
+protocol JPScrollViewDataDelegate {
+
+    var translationY: CGFloat { get }
+    var scrollViewFrameHeight: CGFloat { get }
+    var scrollViewContentHeight: CGFloat { get }
+    var scrollViewContentOffsetY: CGFloat { get }
```

Comment:
Instead of adding all these properties and force every type that conforms to this protocol to implement them, can we just use them in the specific case they are needed? In other words, if there are some viewControllers that just need to send a signal to hide the bar (or simply don't send any signal), can that just be done in `scrollViewDidScroll`?

Reply:
I'm open to that idea, but I wanted to share my thoughts behind it:

We have a very specific protocol for view controllers that are scrollable, `JPScrollViewDelegate`, and before we were just sending the `y` translation data.

This adds a new protocol (`JPScrollViewDataDelegate`) for types to conform to so that they can send more context than just the `y` translation. It stays specific to scrolling views, but allows for us to enhance our decisions, such as not hiding the banner when a scroll view is bouncing.

It would be easy to mock up to test banner visibility if we add tests, and it prevents us from having to repeat the same logic for every view. That was my thinking, at least.

If a view wanted to show or hide the banner _outside_ of the scroll view logic, it'd need to find another way (e.g. just tell the view to hide when rotating to landscape, for instance).